Regression: basic authentication in URL does not work

```
% r2e list  | grep netch
23: [*] netch (https://dottedmag:<redacted password>@netch.dreamwidth.org/data/atom?auth=digest -> dottedmag@dottedmag.net)
% r2e run --no-send
processing error: expected bytes-like object, not str: netch (https://dottedmag:<redacted password>@netch.dreamwidth.org/data/atom?auth=digest -> dottedmag@dottedmag.net)
% r2e --version
rss2email 3.9
%
```

It used to work in an ancient Python 2 version of `rss2email`.

http://test:password@foo.com/bar.php can be used to test this since nothing is sent to the server.

```
[ERROR] processing error: a bytes-like object is required, not 'str': test (http://test:password@foo.com/bar.php -> rss2email)
```

The problem is entirely in feedparser. Encoding user_passwd (in feedparser) fixes the initial error and then throws the same error in `return opener.open(request)`. That code is different in 6.0 so it might have been fixed.

```
-    auth = base64.standard_b64encode(user_passwd).strip()
+    auth = base64.standard_b64encode(user_passwd.encode()).strip()
```

--- a/.github/workflows/python-package.yml
+++ b/.github/workflows/python-package.yml
@@ -28,4 +28,5 @@ jobs:
         poetry install
     - name: Test
       run: |
-        poetry run ./test/test.py -v
+        cd ./test
+        poetry run python -m unittest

--- a/.github/workflows/python-package.yml
+++ b/.github/workflows/python-package.yml
@@ -14,7 +14,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest]
-        python-version: [3.6, 3.7, 3.8]
+        python-version: [3.6, 3.7, 3.8, 3.9]
     steps:
     - uses: actions/checkout@v2
     - name: Set up Python ${{ matrix.python-version }}

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -7,7 +7,7 @@ license = "GPL-2.0-only"
 
 [tool.poetry.dependencies]
 python = "^3.6"
-feedparser = ">=6.0.0"
+feedparser = ">=6.0.5"
 html2text = ">=2020.1.16"
 
 [tool.poetry.dev-dependencies]

--- a/setup.py
+++ b/setup.py
@@ -66,7 +66,7 @@ setuptools.setup(
     scripts=['r2e'],
     provides=['rss2email'],
     install_requires=[
-        'feedparser>=6.0.0',
+        'feedparser>=6.0.5',
         'html2text>=2020.1.16',
         ],
     )

